posthog/api/decide.py

```diff
@@ -42,7 +42,7 @@ def hostname_in_allowed_url_list(allowed_url_list: Optional[List[str]], hostname
 
     for permitted_domain in permitted_domains:
         if "*" in permitted_domain:
-            pattern = "^{}$".format(re.escape(permitted_domain).replace("\\*", "(.*)"))
+            pattern = "^{}$".format(re.escape(permitted_domain).replace("\\*", r"(\w*)"))
```

Comment:
I think this breaks support for sub-subdomains like `app.dev.example.com`, do we have a test for that?

Reply:
Yep, this forbids sub-subdomains intentionally as I found it surprising https://github.com/PostHog/posthog/pull/15557/files#diff-cbd48866da8bccafd5dc1f27e3b4cdb0327842dad31c3dfa3cdbd9e86afab0eaR772. Seems like it's an actual use case though? In that case I'm happy to revert this change.